app/simulation_svc.py

```diff
@@ -98,3 +101,32 @@ def __init__(self, services, agents, loaded_scenario):
                 await self.start_agent(agent)
                 return True
         return False
+
+    """ PRIVATE"""
+
+    async def _extract_vars(self, ability, link):
+        decoded_test = self.decode_bytes(ability.test)
+        decoded_command = self.decode_bytes(link.command)
+        variables = re.findall(r'#{(.*?)}', decoded_test, flags=re.DOTALL)
+        vars = [x for x in variables if x not in ['server', 'paw', 'location', 'group']]
+        search_set = []
+        filtering = decoded_test
+        for v in vars:
+            search_key, filtering = filtering.split("#{" + v + "}", 1)
+            extract = re.findall(r'' + re.escape(search_key) + '(.*)', decoded_command, flags=re.DOTALL)
+            extract = extract[0]
+            if len(filtering) != 0:
+                bound = 3 if len(filtering) > 2 else -1
+                extract = extract.split(filtering[0:bound])[0]
+            search_set.append([v, extract])
+        return search_set
+
+    async def _target_response(self, vars, response_object):
+        computable = dict()
+        for entry in response_object:
+            computable[entry.v_name] = {entry.v_value: [entry.response, entry.status]}
+        for v in vars:
+            if v[0] in computable:
+                if v[1] in computable[v[0]]:
+                    return computable[v[0]][v[1]]
+        return computable['default']['default']
```

Comment:
Can you explain this to me a bit more?  Is the v_name and v_value always going to be Default?